### PR TITLE
Firewall

### PR DESCRIPTION
**Pull request by jdextercsc**

### CODE REVIEW

File: roles/firewall/templates/compute_iptables.j2 [hunk 1]
```diff
@@ -0,0 +1,21 @@
+*filter
+:INPUT ACCEPT [2:104]
+:FORWARD ACCEPT [0:0]
+:OUTPUT ACCEPT [2:204]
+-A INPUT -m state --state ESTABLISHED,RELATED -j ACCEPT
+-A INPUT -p icmp -j ACCEPT
+-A INPUT -i lo -j ACCEPT
+-A INPUT -p tcp -m multiport --dports 22 -m comment --comment "allow ssh incoming" -j ACCEPT
+{% for node in groups[controllers] %}
```
Comment (mtnbikenc) [inline roles/firewall/templates/compute_iptables.j2]: In each 'for' clause you will need single quotes around the group name.  See, https://docs.ansible.com/playbooks_variables.html#magic-variables-and-how-to-access-information-about-other-hosts

File: roles/firewall/templates/compute_iptables.j2 [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- A INPUT -s {{ node[ 'ansible_' + private_if ].ipv4.address }}/32 --dports 4789 -m comment --comment "neutron tunnel port incoming neutron_tunnel_{{ node[ 'ansible_' + private_if ].ipv4.address }}_{{ hostvars[inventory_hostname][ 'ansible_' + private_if ].ipv4.address }}" -j ACCEPT
```
Comment (mtnbikenc) [inline roles/firewall/templates/compute_iptables.j2]: In each case of
`node[ 'ansible_' + private_if ].ipv4.address`
  should be
`hostvars[node][ 'ansible_' + private_if ].ipv4.address`

**Comment by mtnbikenc** [comment]
Several of the templates need to have some corrections in variable usage.  Code comments provided for examples.

**Comment by mtnbikenc** [comment]
Code review :+1:

**Comment by jcpowermac** [comment]
Code review :+1:

**Pull request merged**